bin/setup

```diff
@@ -34,147 +35,148 @@ CONTAINER_TAG=`cat ${ROOT}/bin/.container-tag`
 #  3. Docker, if DOCKER is set in the environment
 #  4. Singularity, if SINGULARITY is set in the environment
 #  5. The choice made by the user, which will be cached in .container-type
-if [ -v DEVCONTAINER_ENV ]; then
-CONTAINER_TYPE=devcontainer
-elif [ -f ${ROOT}/.container-type ]; then
-CONTAINER_TYPE=`cat ${ROOT}/.container-type`
-elif [ -v DOCKER ]; then
-CONTAINER_TYPE=docker
-elif [ -v SINGULARITY ]; then
-CONTAINER_TYPE=singularity
+if [ -n "${DEVCONTAINER_ENV+x}" ]; then
+  CONTAINER_TYPE=devcontainer
+elif [ -f "${ROOT}/.container-type" ]; then
+  CONTAINER_TYPE="$(tr -d '\n' < "${ROOT}/.container-type")"
+elif [ -n "${DOCKER+x}" ]; then
+  CONTAINER_TYPE=docker
+elif [ -n "${SINGULARITY+x}" ]; then
+  CONTAINER_TYPE=singularity
 else
-  echo -e "UDB tools run in a container. Both Docker and Singularity/Apptainer are supported.\\n\\n1. Docker\\n2. Singularity\\n"
+  echo -e "UDB tools run in a container. Both Docker and Singularity/Apptainer are supported.\n\n1. Docker\n2. Singularity\n"
   while true; do
     echo "Which would you like to use? (1/2) "
     read ans
-    case $ans in
-        [1]* ) CONTAINER_TYPE=docker; break;;
-        [2]* ) CONTAINER_TYPE=singularity; break;;
-        * ) echo -e "\\nPlease answer 1 or 2.";;
+    case "$ans" in
+      [1]* ) CONTAINER_TYPE=docker; break;;
+      [2]* ) CONTAINER_TYPE=singularity; break;;
+      * ) echo -e "\nPlease answer 1 or 2.";;
     esac
   done
 fi
 
-if [ "${CONTAINER_TYPE}" != "docker" -a "${CONTAINER_TYPE}" != "singularity" -a "${CONTAINER_TYPE}" != "devcontainer" ]; then
+if [ "$CONTAINER_TYPE" != "docker" -a "$CONTAINER_TYPE" != "singularity" -a "$CONTAINER_TYPE" != "devcontainer" ]; then
   echo "BAD CONTAINER TYPE: ${CONTAINER_TYPE}"
 fi
 
-if [ ! -f ${ROOT}/.container-type ]; then
-  echo ${CONTAINER_TYPE} > ${ROOT}/.container-type
+if [ ! -f "${ROOT}/.container-type" ]; then
+  echo "${CONTAINER_TYPE}" > "${ROOT}/.container-type"
```

Comment:
You removed the braces above. I don't think we need them here, either. We should probably pick a consistent style.

Reply:
@ThinkOpenly  this is on picking one style (with or without braces) and use it consistently like the previous case.

Reply:
My vote would be either:
- no innocuous changes, or
- make all the innocuous changes needed for consistency

Having some, but not all, makes it challenging to review. "Why is this changed, but that isn't?"